### PR TITLE
fix!: remove GovPreventSpamDecorator

### LINKED ISSUE #2911: GovPreventSpamDecorator is no longer needed
# Problem

SDK 0.47 introduced SDK 0.47 introduced the `MinInitialDepositRatio` param to the gov module (see https://github.com/cosmos/cosmos-sdk/pull/12771). This means we no longer need the `GovPreventSpamDecorator` in Gaia. The same check is done when handling a `SubmitProposal` message, see [here](https://github.com/cosmos/cosmos-sdk/blob/v0.47.8/x/gov/keeper/msg_server.go#L32).

# Closing criteria

- Remove `GovPreventSpamDecorator`
- Initialize the `MinInitialDepositRatio` param of the gov module to `sdk.NewDecWithPrec(10, 2)`, see https://github.com/cosmos/cosmos-sdk/blob/v0.47.8/proto/cosmos/gov/v1/gov.proto#L210

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: #2911

Remove `GovPreventSpamDecorator` as this logic is already implemented in SDK 0.47, see https://github.com/cosmos/cosmos-sdk/blob/v0.47.8/x/gov/keeper/msg_server.go#L32. 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if API, client, or state breaking change (i.e., requires minor or major version bump)
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] Provided a link to the relevant issue or specification
* [x] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)
* [x] Added a changelog entry in `.changelog` (for details, see [contributing guidelines](../../CONTRIBUTING.md#changelog))
* [x] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [x] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed all author checklist items have been addressed 
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: types/errors/errors.go [hunk 1]
```diff
@@ -14,21 +14,18 @@ var (
 	ErrUnauthorized = errorsmod.Register(codespace, 2, "unauthorized")
 
 	// ErrInsufficientFunds is used when the account cannot pay requested amount.
-	ErrInsufficientFunds = errorsmod.Register(codespace, 3, "insufficient funds")
```
Comment (MSalopek) [inline types/errors/errors.go]: Don't change error codes and don't re-map error codes. Treat this list as append only.

The err codes may be written alongside the failed Tx in the state (then querying `gaiad q tx <hash>`). Changing these codes could result in incosistent parsing or incorrect interpretation.

Cosmos-sdk example with legacy errors:
* https://github.com/cosmos/cosmos-sdk/blob/main/x/gov/types/errors.go#L11-L13

**Review by MSalopek — CHANGES_REQUESTED** [review]
Don't remap errors, other than that LGTM

File: app/upgrades/v15/upgrades.go [hunk 2]
```diff
@@ -323,3 +327,21 @@ func setBalance(
 
 	return nil
 }
+
+// SetMinInitialDepositRatio sets the MinInitialDepositRatio param of the gov
+// module to 10% - this is the proportion of the deposit value that must be paid
+// at proposal submission.
+func setMinInitialDepositRatio(ctx sdk.Context, gk govkeeper.Keeper) error {
```
Comment (MSalopek) [inline app/upgrades/v15/upgrades.go]: Write a small test to double check this. Check `TestUpgradeMinCommissionRate` (or any other tests, but this one already changes with params).

File: types/errors/errors.go (hunk 1 above)
Comment (mpoke) [reply types/errors/errors.go]: See https://github.com/cosmos/gaia/pull/2913/commits/dc6fd2b94c31460d82cdc8866ea9f712e57efdb4

File: app/upgrades/v15/upgrades.go (hunk 2 above)
Comment (mpoke) [reply app/upgrades/v15/upgrades.go]: See https://github.com/cosmos/gaia/pull/2913/commits/c7a7229fe62e0ec0fb7984de2119b57b8af7c406

File: app/upgrades/v15/upgrades_test.go [hunk 3]
```diff
@@ -270,3 +270,15 @@ func TestClawbackVestingFunds(t *testing.T) {
 	)
 	require.NoError(t, err)
 }
+
+func TestSetMinInitialDepositRatio(t *testing.T) {
+	gaiaApp := helpers.Setup(t)
+	ctx := gaiaApp.NewUncachedContext(true, tmproto.Header{})
+
+	v15.SetMinInitialDepositRatio(ctx, *gaiaApp.GovKeeper)
```
Comment (MSalopek) [inline app/upgrades/v15/upgrades_test.go]:
```suggestion
	err := v15.SetMinInitialDepositRatio(ctx, *gaiaApp.GovKeeper)
	require.NoError(t, err)
```

**Review by MSalopek — APPROVED** [review]
LGTM!

Please appease linter before merge.

**Review by sainoe — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**